--- a/package/openwrt/files/etc/config/smartdns	2020-08-20 00:44:00.000000000 +0800
+++ b/package/openwrt/files/etc/config/smartdns	2022-04-04 11:29:17.818078994 +0800
@@ -1,3 +1,26 @@
-config 'smartdns'
-	option 'enabled' '0'
-	
\ No newline at end of file
+
+config smartdns
+	option enabled '1'
+	option server_name 'smartdns'
+	option tcp_server '1'
+	option prefetch_domain '1'
+	option serve_expired '1'
+	option redirect 'dnsmasq-upstream'
+	option ipv6_server '1'
+	option port '6053'
+	option rr_ttl_min '300'
+	option force_aaaa_soa '0'
+	option coredump '0'
+	option cache_size '65535'
+	option dualstack_ip_selection '1'
+	option seconddns_enabled '1'
+	option seconddns_server_group 'oversea'
+	option seconddns_no_cache '0'
+	option seconddns_tcp_server '1'
+	option seconddns_port '8653'
+	option seconddns_no_rule_ipset '1'
+	option seconddns_no_speed_check '0'
+	option seconddns_no_rule_addr '1'
+	option seconddns_no_rule_nameserver '1'
+	option seconddns_no_dualstack_selection '0'
+	option seconddns_no_rule_soa '1'
